core/src/main/scala/za/co/absa/spline/harvester/conf/Spline05ConfigurationAdapter.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.conf
+
+import org.apache.commons.configuration.Configuration
+import za.co.absa.spline.harvester.conf.DefaultSplineConfigurer.ConfProperty._
+import za.co.absa.spline.harvester.conf.Spline05ConfigurationAdapter._
+import za.co.absa.spline.harvester.dispatcher.httpdispatcher.HttpLineageDispatcherConfig._
+import za.co.absa.spline.harvester.iwd.DefaultIgnoredWriteDetectionStrategy._
+
+import java.util
+import scala.collection.JavaConverters.asJavaIterableConverter
+
+class Spline05ConfigurationAdapter(configuration: Configuration) extends ReadOnlyConfiguration {
+
+  override def isEmpty: Boolean = scalaKeys.isEmpty
+
+  override def containsKey(key: String): Boolean = keyMap.get(key).map(configuration.containsKey).getOrElse(false)
+
+  override def getProperty(key: String): AnyRef = keyMap.get(key).map(configuration.getProperty).getOrElse(null)
```

Comment:
When is called with `spline.lineageDispatcher` as a key, it's supposes to return a default dispatcher name (in this case it would `http`). But it should return it if and only if the `spline.lineage_dispatcher.className` is specified. That is because the "old" entry `spline.lineage_dispatcher.className=FooBar` is represented by two "new" entries: `spline.lineageDispatcher=something` + `spline.lienageDispatcher.something.className=FooBar`. So in reverse, if the config source reports that `spline.lienageDispatcher.something.className` is defined, then by definition the complimentary property `spline.lienageDispatcher` with the value.`something` has to also be defined.

Reply:
I think this class wants some unit tests.